### PR TITLE
Add Carthage support

### PR DESCRIPTION
**Pull request by Palleas**
Hello, 

I've added Carthage support for this project I started using. I know it says "Carthage compatible" but it 
actually didn't provide a shared Framework scheme. This is the error you'd get if you tried using it: 

```
$  carthage update
*** Fetching SwiftGif
*** Checking out SwiftGif at "v1.5.1"
*** xcodebuild output can be found in /var/folders/tk/cw93k7v928qf21kppt47d_380000gn/T/carthage-xcodebuild.QXYO0e.log
*** Skipped building SwiftGif due to the error:
Dependency "SwiftGif" has no shared framework schemes

If you believe this to be an error, please file an issue with the maintainers at https://github.com/bahlo/SwiftGif/issues/new
```

Here is what I've done: 
* Moved the code from "SwiftGifCommon" into a dedicated "SwiftGif" folder and created the framework scheme
* Renamed the "SwiftGif" target into "SwiftGifDemo" 
* I updated the path in the podspec as well and bumped the version number so you can tag /  push

Hopefully the travis build should still pass. Let me know what you think about this :smile:

### CODE REVIEW

**Comment by bahlo** [comment]
Thank you very much! I will look into this :+1:

**Comment by Palleas** [comment]
The Travis build is failing though, not sure why because I run the same command on my mac and it works :cry:

File: .travis.yml [hunk 1]
```diff
@@ -4,5 +4,5 @@ osx_image: xcode7
 script:
   - xctool
     -project SwiftGif.xcodeproj -scheme SwiftGif
-    -sdk iphonesimulator build test
+    -sdk iphonesimulator -destination "platform=iOS Simulator,name=iPhone 6" build test
```
Comment (bahlo) [inline .travis.yml]: I think this is the line that is causing Travis to fail. When I run this command on my Mac, the tests still pass, but only after I get the following warning:

```
[Warning] Tried to install the test host app 'me.arne.SwiftGif' but failed. (3132 ms)
[Info] Preparing test environment failed; will retry 2 more times (0 ms)
[Info] Verified iOS Simulators... (0 ms)
[Info] Shut down iOS Simulator... (0 ms)
[Info] Stopped any existing iOS simulator jobs to get a fresh simulator. (364 ms)
[Info] Reset iOS simulator content and settings at path ...
```

Travis is failing at the same step: https://travis-ci.org/bahlo/SwiftGif/builds/93796224#L202

**Comment by bahlo** [comment]
I also saw [your tweet](https://twitter.com/Palleas/status/671017954763501569) and I'm curious and happy to hear what kind of application you use this in (if you can disclose). :relaxed:

File: .travis.yml (hunk 1 above)
Comment (Palleas) [inline .travis.yml]: I may have broken something when I moved target around. Didn't notice the warning, thought. 
I'll have a look, thanks :smile:

**Comment by Palleas** [comment]
I'm using it to render Gif in an iOS extension, I'll put it on Github when it's done!

**Comment by bahlo** [comment]
Merged in [v1.6-beta.1](https://github.com/bahlo/SwiftGif/releases/tag/v1.6-beta.1) (soon to be master) thank you so much 👍

**Pull request closed without merging**